pallets/mips/src/lib.rs

```diff
@@ -640,37 +679,70 @@ decl_module! {
                 Error::<T>::NotACommitteeMember
             );
 
-            let mip = Self::proposals(index)
-                .ok_or(Error::<T>::MismatchedProposalIndex)?;
+            ensure!(Self::proposals(index).is_some(), Error::<T>::MismatchedProposalIndex);
 
             Self::create_referendum(
                 index,
-                MipsPriority::High,
-                mip.proposal,
+                ReferendumState::Pending,
+                ReferendumType::FastTracked,
             );
 
-            Self::deposit_event(RawEvent::ProposalFastTracked(index));
-            Self::close_proposal(index);
+            Self::update_proposal_state(index, ProposalState::Referendum);
+
+            // Update proposal metadata so we don't re-execute it later
+            <ProposalMetadata<T>>::mutate( |metas| {
+                let meta = metas.iter_mut().find( |meta| meta.index == index);
```

Comment:
Unbounded storage access. Ok for testnet v1, should revisit for v2. Perhaps add a TODO?

Reply:
Added TODO comment

Reply:
BTW, It has been fixed in [PR 303](https://github.com/PolymathNetwork/Polymesh/pull/303/)